File: app/src/main/java/di/penopllast/com/inforemover/entities/ApplicationAdapter.java
package di.penopllast.com.inforemover.entities;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;


import java.util.List;

import di.penopllast.com.inforemover.R;

public class ApplicationAdapter extends RecyclerView.Adapter {

    private final List<ApplicationInfo> appList;

    public ApplicationAdapter(List<ApplicationInfo> appList) {
        this.appList = appList;
        Log.i("test", "size = " + appList.size());
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public final ImageView icon;
        public final TextView name;

        public ViewHolder(View view) {
            super(view);
            this.icon = view.findViewById(R.id.icon);
            this.name = view.findViewById(R.id.name);
        }
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_app, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Log.i("test", "position = " + position);
        ApplicationInfo appInfo = appList.get(position);
        ((ViewHolder) holder).name.setText(appInfo.getName());
        ((ViewHolder) holder).icon.setImageDrawable(appInfo.getIcon());
    }

    @Override
    public int getItemCount() {
        return appList.size();
    }
}

File: app/src/main/java/di/penopllast/com/inforemover/entities/ApplicationAdapter.kt
package di.penopllast.com.inforemover.entities

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import di.penopllast.com.inforemover.R
import kotlinx.android.synthetic.main.item_app.view.*
import di.penopllast.com.inforemover.application.Utils.Util


class ApplicationAdapter(private val appList: List<ApplicationInfo>)
    : RecyclerView.Adapter<ApplicationAdapter.ViewHolder>() {

    init {
        Log.i("test", "size = " + appList.size)
    }

    class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
        val icon = view.icon_image!!
        val name = view.name_text!!
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_app, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Log.i("test", "position = $position")
        val appInfo = appList[position]
        holder.name.text = appInfo.name
        holder.icon.setImageDrawable(appInfo.icon)
        holder.view.setOnClickListener { Util.print("Нажали на ${appInfo.name}")}
    }

    override fun getItemCount(): Int {
        return appList.size
    }
}
